contracts/JBPaymentTerminal.sol

```diff
@@ -0,0 +1,987 @@
+// SPDX-License-Identifier: MIT
+/* solhint-disable comprehensive-interface*/
+pragma solidity 0.8.6;
+
+import '@openzeppelin/contracts/token/ERC20/IERC20.sol';
+import '@paulrberg/contracts/math/PRBMath.sol';
+
+import './libraries/JBConstants.sol';
+import './libraries/JBCurrencies.sol';
+import './libraries/JBOperations.sol';
+import './libraries/JBSplitsGroups.sol';
+import './libraries/JBTokens.sol';
+
+import './JBPaymentTerminalStore.sol';
+
+// Inheritance
+import './interfaces/IJBPaymentTerminal.sol';
+import './interfaces/IJBTerminal.sol';
+import './abstract/JBOperatable.sol';
+import '@openzeppelin/contracts/access/Ownable.sol';
+import '@openzeppelin/contracts/security/ReentrancyGuard.sol';
```

Comment:
Reentrancy is not a risk here (no eth and project wanting to use erc777 with a onReceive hook for instance would use another terminal)

Reply:
JBPaymentTerminal is abstract and used by both the ETH terminal and the ERC ones

Reply:
neeat